symengine/expression.h

```diff
@@ -32,16 +32,16 @@ class Expression
     //! Construct Expression from `int`
     Expression(int n) : m_basic(integer(n)) {}
     //! Construct Expression from Basic
-    template <typename T, typename = typename std::enable_if<std::is_constructible<RCP<const Basic>, T &&>::value>::type>
+    template <typename T>
     Expression(T &&o) : m_basic(std::forward<T>(o)) {}
     //! Construct Expression from Expression
     Expression(const Expression &) = default;
     //! Construct Expression from reference to Expression
-    Expression(Expression &&other) noexcept : m_basic(std::move(other.m_basic)) {}
```

Comment:
The Intel compiler doesn't seem to understand the `noexcept` keyword, so a workaround might be to use a macro `NOEXCEPT` and leave it empty for Intel.